common4j/src/main/com/microsoft/identity/common/java/controllers/BaseController.java

```diff
@@ -733,7 +790,17 @@ protected TokenResult performSilentTokenRequest(
         }
 
         final TokenRequest refreshTokenRequest = strategy.createRefreshTokenRequest(parameters.getAuthenticationScheme());
-        refreshTokenRequest.setClientId(parameters.getClientId());
+
+        if (parameters.hasNestedAppParameters()) {
+            // isNAA request, set hub/brk and nested app parameters
+            refreshTokenRequest.setClientId(parameters.getChildClientId());
+            refreshTokenRequest.setBrkClientId(parameters.getClientId());
```

Comment:
I believe Brkr here is the hub like Teams Native app? and client id (child) is the embedded web app?

Reply:
yes child app can be any Single page app for now

Reply:
Not a being fan of the parameter naming here. The `Brkr` to me means the Android Broker, but over here I believe these are the hub apps. Should these be renamed to `HubClientId`?

Reply:
My understanding is that brkr is what ests uses... but I do agree that it's confusing given our context.

Reply:
I could change this to HubClientId but it would be inconsistent with ESTS's convention. They call it the brkClientId.